t/lib/WTSI/NPG/iRODS/icommandsTest.pm

```diff
@@ -26,18 +27,19 @@ sub list_specific_query_test : Test(1) {
   ok(iquest('--sql', 'ls'), 'Can list specific queries');
 }
 
-sub single_placeholder_test : Test(1) {
+sub single_placeholder_test : Test(7) {
   my ($self) = @_;
 
   my @colls = iquest(qw[%s], q[select COLL_NAME where COLL_NAME like '/testZone/%']);
-  is_deeply(\@colls,
-     ['/testZone/home',
-      '/testZone/home/irods',
-      '/testZone/home/public',
-      '/testZone/trash',
-      '/testZone/trash/home',
-      '/testZone/trash/home/irods',
-      '/testZone/trash/home/public']) or diag explain \@colls;
+  for my $c ( qw( /testZone/home
+                  /testZone/home/irods
+                  /testZone/home/public
+                  /testZone/trash
+                  /testZone/trash/home
+                  /testZone/trash/home/irods
+                  /testZone/trash/home/public ) ) {
+    ok ((any { $_ =~ /\A$c\Z/ } @colls), "$c is listed");
```

Comment:
What was the reason for changing this test? It seems unrelated to the rest of the changes.

Reply:
(From the commit message: it's to work around the tests failing if there's state left from previous rounds of testing.)

The tests should start with iRODS in a known state. The cleanest way to do that is to start a new server, or to make a more effective clean-up method (or script) to remove everything and ensure that it is run before the suite. (The suite already tries to clean up anything it creates).

I don't agree that changing individual tests is a good thing; it addresses only that test and there are many others which haven't been hardened to work around unexpected state on the server.

Reply:
Most likely other unit tests are not as sensitive as this one because they look at collections and objects which are created at a particular location for the purpose of the particular test.

Reply:
Anything involving metadata is sensitive to global state (within the same zone).

Reply:
I'm not going to block this, please feel free to merge.

Reply:
Dropped the commit